### PR TITLE
Add gitpod support#604 

### PR DESCRIPTION
**Pull request by khan-asfi-reza**
# Gitpod support

## Description
Added gitpod support ( Issue #604 )

### CODE REVIEW

File: .gitpod.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM gitpod/workspace-full
+
+USER gitpod
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+
+RUN rustup default stable
```
Comment (vil02) [inline .gitpod.Dockerfile:7]: Please note that there is an [official and dedicated rust _gitpod_ image](https://hub.docker.com/r/gitpod/workspace-rust). 

```suggestion
FROM gitpod/workspace-rust:2023-11-16-11-19-36

USER gitpod

```

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+image:
+  file: .gitpod.Dockerfile
+
+tasks:
+  - init: rustup update && rustup component add rustfmt clippy
+    command: cargo build
```
Comment (vil02) [inline .gitpod.yml:6]: Is this needed?

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+vscode:
+  extensions:
+    - rust-lang.rust
```
Comment (vil02) [inline .gitpod.yml:10]: When I open gitpod, I get a notification that this extension is deprecated.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: .gitpod.yml (hunk 2 above)
Comment (khan-asfi-reza) [reply .gitpod.yml]: Please check the new commit

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**